### PR TITLE
Add support to send custom parameters/headers in the subscription request

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes: [#4795](https://github.com/ballerina-platform/ballerina-standard-library/issues/4795)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [x] Updated the changelog
- [X] Added tests
- [x] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/commons.bal [hunk 1]
```diff
@@ -113,15 +113,19 @@ public type ListenerConfiguration record {|
...
+# + secret - The secret to be used for authenticated content distribution with this subscription  
+# + additionalParams - Additional parameters which needs to be sent with the subscription request
+# + additionalHeaders - Additional HTTP headers which needs to be sent with the subscription request
 public type SubscriptionChangeRequest record {|
     string topic = "";
     string callback = "";
     int leaseSeconds = 0;
     string secret = "";
+    map<string> additionalParams?;
+    map<string> additionalHeaders?;
```
Comment (shafreenAnfar) [inline ballerina/commons.bal:128]: Lets use the prefix custom

Comment (ayeshLK) [reply ballerina/commons.bal]: Fixed with [a30a78cbb6c24bd436a9b87f08815fb52f46e6c7](https://github.com/ballerina-platform/module-ballerina-websub/pull/629/commits/a30a78cbb6c24bd436a9b87f08815fb52f46e6c7)

File: ballerina/annotation.bal [hunk 2]
```diff
@@ -18,15 +18,17 @@ import ballerina/http;
...
+# or a tuple `[hub, topic]` representing a discovered hub and a topic  
+# + leaseSeconds - The period for which the subscription is expected to be active  
+# + callback - The callback URL for subscriber-service  
+# + secret - The secret to be used for authenticated content distribution  
+# + appendServicePath - This flag notifies whether or not to append service-path to callback-url  
+# + unsubscribeOnShutdown - This flag notifies whether or not to initiate unsubscription when the service is shutting down  
+# + httpConfig - The configuration for the hub client used to interact with the discovered/specified hub  
+# + discoveryConfig - HTTP client configurations for resource discovery  
+# + customParams - Additional parameters which needs to be sent with the subscription request
+# + customHeaders - Additional HTTP headers which needs to be sent with the subscription request
```
Comment (NipunaRanasinghe) [inline ballerina/annotation.bal:31]:
```suggestion
# + customParams - Additional parameters which need to be sent with the subscription request
# + customHeaders - Additional HTTP headers which need to be sent with the subscription request
```

File: ballerina/commons.bal [hunk 3]
```diff
@@ -113,15 +113,19 @@ public type ListenerConfiguration record {|
...
-# + topic - The topic for which the subscription/unsubscription request is sent
-# + callback - The callback which should be registered/unregistered for the subscription/unsubscription request sent
-# + leaseSeconds - The lease period for which the subscription is expected to be active
-# + secret - The secret to be used for authenticated content distribution with this subscription
+# + topic - The topic for which the subscription/unsubscription request is sent  
+# + callback - The callback which should be registered/unregistered for the subscription/unsubscription request sent  
+# + leaseSeconds - The lease period for which the subscription is expected to be active  
+# + secret - The secret to be used for authenticated content distribution with this subscription  
+# + customParams - Additional parameters which needs to be sent with the subscription request
+# + customHeaders - Additional HTTP headers which needs to be sent with the subscription request
```
Comment (NipunaRanasinghe) [inline ballerina/commons.bal:121]:
```suggestion
# + customParams - Additional parameters which need to be sent with the subscription request
# + customHeaders - Additional HTTP headers which need to be sent with the subscription request
```

File: docs/spec/spec.md [hunk 4]
```diff
@@ -279,6 +279,8 @@ should be `websub:SubscriberServiceConfig` a service-level-annotation for `websu
 # + unsubscribeOnShutdown - This flag notifies whether or not to initiate unsubscription when the service is shutting down
 # + httpConfig - The configuration for the hub client used to interact with the discovered/specified hub
 # + discoveryConfig - HTTP client configurations for resource discovery
+# + customParams - Additional parameters which needs to be sent with the subscription request
+# + customHeaders - Additional HTTP headers which needs to be sent with the subscription request
```
Comment (NipunaRanasinghe) [inline docs/spec/spec.md:283]:
```suggestion
# + customParams - Additional parameters which need to be sent with the subscription request
# + customHeaders - Additional HTTP headers which need to be sent with the subscription request
```

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**